npep/npep-137-conformance-profiles.md

```diff
@@ -0,0 +1,187 @@
+# NPEP-137: Conformance Profiles
+
+* NOTE: Adapted with love from https://gateway-api.sigs.k8s.io/geps/gep-1709/ (Then
+a question may arise, why do we need our own NPEP? Answer: Gateway API Project has
+lots of resources and features and it gets very complicated to adopt their same API.
+We need to tune the profiles to be compatible with what works for Network Policy API
+project, thus the need for this NPEP). Our profiles are much simpler than what GatewayAPI
+supports
+* Issue: [#137](https://github.com/kubernetes-sigs/network-policy-api/issues/137)
+* Status: Implementable
+
+## TLDR
+
+Add conformance profiles for existing conformance tests which implementations
+can leverage when running the conformance test suite in their downstream repos
+to report the results of the tests back to the Network Policy API project and
+receive recognition (eg. "conformance badge")
+
+## Goals
+
+* Add conformance profiles for existing tests that downstream implementations
+can subscribe to to run tests associated to the supported API feature sets
+
+* Add a reporting mechanism via a new CRD where conformance results can be
+reported back to the Network Policy API project and provide "conformance badges"
+via Network Policy official website to recognize these implementations.
+
+* Expand existing conformance testing framework to account for profiles
+
+## Non-Goals
+
+* The first iteration will just have a simple repo under which the report
+results will be stored, no automated reporting infrastructure will be built for this now.
+Implementations will need to open PRs against network-policy-api repo to upload
+the results.
+
+## Introduction
+
+Currently, the conformance tests are grouped into `StandardCoreFeatures` and
+`StandardExtendedFeatures`. The features that fall under the `StandardCoreFeatures`
+are a must for the implementations to be conformant while the features that fall
+under `StandardExtendedFeatures` are opt-ins which are not mandatory.
+
+In this NPEP, we will add a concept called `named profiles` which will indicate a
+"level of conformance" on a resource level. We are picking profiles on a resource
+level because that is what makes sense for the project as of today. This can be
```

Comment:
```suggestion
In this NPEP, we will add a concept called `named profiles` which will indicate a
"level of conformance" on for a given resource. The initial decision to report conformance profiles on a per-resource level  is a choice which may be re-evaluated by the working group in the future.
This can be
```